doc/source/developing/testing.rst

```diff
@@ -3,503 +3,386 @@
-module.  Describing them in detail is somewhat outside the scope of this
-document, as in some cases they belong to other packages.  However, a few come
-in handy:
-
-* :func:`~yt.testing.fake_random_ds` provides the ability to create a random
-  dataset, with several fields and divided into several different
-  grids, that can be operated on.
-* :func:`~yt.testing.assert_equal` can operate on arrays.
-* :func:`~yt.testing.assert_almost_equal` can operate on arrays and accepts a
-  relative allowable difference.
-* :func:`~yt.testing.assert_allclose_units` raises an error if two arrays are
-  not equal up to a desired absolute or relative tolerance. This wraps numpy's
-  assert_allclose to correctly verify unit consistency as well.
-* :func:`~yt.testing.amrspace` provides the ability to create AMR grid
-  structures.
-* :func:`~yt.testing.expand_keywords` provides the ability to iterate over
-  many values for keywords.
+yt provides several helper functions and decorators to write unit tests. These tools all reside in the ``yt.testing``
+module.  Describing them all in detail is outside the scope of this
+document, as in some cases they belong to other packages.
+
+How To Write New Unit Tests
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 To create new unit tests:
 
 #. Create a new ``tests/`` directory next to the file containing the
    functionality you want to test and add an empty ``__init__.py`` file to
    it.
-#. Inside that directory, create a new python file prefixed with ``test_`` and
-   including the name of the functionality.
-#. Inside that file, create one or more routines prefixed with ``test_`` that
-   accept no arguments. The test function should do some work that tests some
+   #. If a ``tests/`` directory already exists, there is no need to create a new one.
+#. Inside this new ``tests/`` directory, create a new python file prefixed with ``test_`` and
+   including the name of the functionality or source file being tested.
+   #. If a file testing the functionality you're interested in already exists, please add your tests to the existing file.
+#. Inside this new ``test_`` file, create one or more routines prefixed with ``test_`` that
+   accept no arguments.
+#. Each test function should do some work that tests some
    functionality and should also verify that the results are correct using
    assert statements or functions.
-#. Tests can ``yield`` a tuple of the form ``function``, ``argument_one``,
-   ``argument_two``, etc.  For example ``yield my_test, 'banana', 2.0`` would be
-   captured by nose and the ``my_test`` function will be run with the provided
-   arguments.
-#. Use ``fake_random_ds`` to test on datasets, and be sure to test for
-   several combinations of ``nproc``, so that domain decomposition can be
+#. If a dataset is needed, use ``fake_random_ds``, ``fake_amr_ds``, or ``fake_particle_ds`` (the former two of which have a particle ``True``/``False`` option that may be utilized) and be sure to test for
```

Comment:
I think this is erroneous: `fake_random_ds` and `fake_amr_ds` have a `particles` args but it looks from the signatures that it's supposed to be an int, while `fake_particle_ds` interestingly has no such arg